### PR TITLE
Add riscv64 support

### PR DESCRIPTION
**Pull request by U2FsdGVkX1**
tested in RISC-V 64.

please correct me if any problems

### CODE REVIEW

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Comment by VladimirSlavik** [comment]
Can you please add also a release note? Make a copy of `docs/release-notes/template.rst` and fill it in.

**Review by vojtechtrefny — APPROVED** [review]

**Comment by U2FsdGVkX1** [comment]
> Unit tests also need some adjusting...
> 
> ```
> =========================== short test summary info ============================
> FAILED unit_tests/pyanaconda_tests/modules/storage/test_platform.py::PlatformTestCase::test_efi
> FAILED unit_tests/pyanaconda_tests/modules/storage/test_platform.py::PlatformTestCase::test_unsupported_platform
> ====== 2 failed, 2029 passed, 1 xfailed, 18 warnings in 295.86s (0:04:55) ======
> FAIL unit_tests/unit_tests.sh (exit status: 1)
> ```

I have added test and release note, please recheck, thanks!

File: docs/release-notes/riscv64-support.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+:Type: Platform
+:Summary: Add RISC-V 64 support
+
+:Description:
+    In storage stage, partitions cannot be completed
+    and maybe UEFI is not recognized
```
Comment (VladimirSlavik) [inline docs/release-notes/riscv64-support.rst:6]: Um, so this is not really support, but only some part of it?

Comment (M4rtinK) [reply docs/release-notes/riscv64-support.rst:6]: Hmm, maybe we need someone from the non-mainstream architecture are on Fedora to chime in ? CCing @rwmjones and @nullr0ute

Comment (rwmjones) [reply docs/release-notes/riscv64-support.rst:6]: I'm not at all clear on what this comment in the documentation means, but I can say that RISC-V does support partitions (obviously!) and that there is at least preliminary UEFI support although it is not widely used.

Comment (davidlt) [reply docs/release-notes/riscv64-support.rst:6]: We do run some Pungi composes that. That does incl. booting riscv64 VM with EDK2 and running anaconda. I have a patch 1+ week old. Let me cross-check.

File: pyanaconda/modules/storage/bootloader/efi.py [hunk 2]
```diff
@@ -279,6 +279,15 @@ def __init__(self):
         self._is_32bit_firmware = True
 
 
+class RISCV64EFIGRUB(EFIGRUB):
+    _serial_consoles = ["ttyS"]
+    _efi_binary = "\\grubriscv64.efi"
+
+    def __init__(self):
+        super().__init__()
+        self._packages64 = []
```
Comment (davidlt) [inline pyanaconda/modules/storage/bootloader/efi.py:288]: This should be `self._packages64 = ["grub2-efi-riscv64"]`

**Comment by davidlt** [comment]
In Fedora/RISCV Koji I mainly have `EFI` variant, and that code looks almost exactly the same minus one line. This part in Pungi composes succeeds (i.e. anaconda does not fail), but still don't have full Pungi phases passing to verify the final product.

LGTM

File: docs/release-notes/riscv64-support.rst (hunk 1 above)
Comment (U2FsdGVkX1) [reply docs/release-notes/riscv64-support.rst:6]: > Um, so this is not really support, but only some part of it?

No, I mean, before that anaconda couldn't be partitioned, now it's fine (maybe I should rewrite the release note)
and UEFI I need more test.

**Comment by U2FsdGVkX1** [comment]
Hi, are you there?

**Pull request merged**